src/models/IDPDiscovery.js

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+  'okta',
+  './BaseLoginModel',
+  'util/CookieUtil',
+  'util/Enums',
+  '../shared/util/Util'
+],
+function (Okta, BaseLoginModel, CookieUtil, Enums, Util) {
+
+  var _ = Okta._;
+
+  return BaseLoginModel.extend({
+
+    props: function () {
+      var settingsUsername = this.settings && this.settings.get('username'),
+          rememberMeEnabled = this.settings && this.settings.get('features.rememberMe'),
+          cookieUsername = CookieUtil.getCookieUsername(),
+          remember = false,
+          self = this,
+          username;
+      if (settingsUsername) {
+        username = settingsUsername;
+        remember = rememberMeEnabled && username === cookieUsername;
+      }
+      else if (rememberMeEnabled && cookieUsername) {
+        // Only respect the cookie if the feature is enabled.
+        // Allows us to force prompting when necessary, e.g. SAML ForceAuthn
+        username = cookieUsername;
+        remember = true;
+      }
+
+      return {
+        username: {
```

Comment:
Should this be changed to say 'email'? I left it as is right now, because username is tied to the Security Beacon logic, and technically a username can be email for primary auth. Please let me know what you think